ts: Use scheduling group to do disk I/O

Lazin opened the pull request:
Currently, the ts_read_sg consumer group is only used for segment hydration. This allows the reader that consumes from the cloud storage cache to saturate the disk t-put. This commit fixes that by shifting stream creation operation to also run under the scheduling group.

<!--
See https://github.com/redpanda-data/redpanda/blob/dev/CONTRIBUTING.md#pull-request-body
for more details and examples of what is expected in a PR body.

Content in this top section is REQUIRED. Describe, in plain language, the motivation
behind the change (bug fix, feature, improvement) in this PR and how the included
commits address it.

Add the GitHub keyword `Fixes` to link to bug(s) this PR will fix, e.g.
  Fixes #ISSUE-NUMBER, Fixes #ISSUE-NUMBER, ...

If this PR is a backport, link to the original with `Backport of PR`, e.g.
  Backport of PR #PR-NUMBER
-->

## Backports Required

<!-- Checking at least one of the checkboxes is REQUIRED if this PR is not a backport. -->

- [ ] none - not a bug fix
- [ ] none - this is a backport
- [ ] none - issue does not exist in previous branches
- [ ] none - papercut/not impactful enough to backport
- [x] v25.2.x
- [ ] v25.1.x
- [ ] v24.3.x
- [ ] v24.2.x

## Release Notes

<!--
If the changes in this PR do not need to be mentioned in the release
notes, then don't add a sub-section and simply list `none`, e.g.

* none

Otherwise, adding a sub-section or `none` is REQUIRED if the PR is not a backport PR.
If this is a backport PR, adding contents to this section will override
the release notes section inherited from the original PR to dev.

Add one or more of the sub-sections with a short description bullet
point of the change, e.g.

### Bug Fixes

* Short description of the bug fix if this is a PR to `dev` branch.

### Features

* Short description of the feature. Explain how to configure.

### Improvements

* Short description of how this PR improves existing behavior.

-->
* none

StephanDollberg commented:
From my testing this still leaves the reads in `fetch`. 

I think this is because here we are just opening the streams in the `ts_read` group but really what we need to do is do the dma_reads in the `ts_read` group? I don't think the creation matters really.

vbotbuildovich commented:
<!--TEST RESULTS-->
<!--b0bd6240-f9b6-477e-b576-9a0813353b51-->
<!--TEST RESULTS-->
<!--9030d2cb-3cee-4e23-b3c7-22a0a5018863-->
<!--title-->
## CI test results
<!--/title-->
<!--body-->
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/69333">build#69333</a></summary>


| test_class               | test_method          | test_arguments                                                                                                                           | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                                |
|:-------------------------|:---------------------|:-----------------------------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:----------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| RandomNodeOperationsTest | test_node_operations | {"cloud_storage_type": 1, "compaction_mode": "sliding_window", "enable_failures": false, "mixed_versions": false, "with_iceberg": false} | ducktape    | https://buildkite.com/redpanda/redpanda/builds/69333#01982ce6-0939-4919-a54c-39efeeb02a15 | FLAKY         | 20/21    | upstream reliability is '99.1769547325103'. current run reliability is '95.23809523809523'. drift is 3.93886 and the allowed drift is set to 50. The test should PASS |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/69353">build#69353</a></summary>


| test_class                         | test_method                   | test_arguments                                                                                                                         | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                                  |
|:-----------------------------------|:------------------------------|:---------------------------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| ArchivalTest                       | test_connection_flicker       | {"cloud_storage_type": 1}                                                                                                              | ducktape    | https://buildkite.com/redpanda/redpanda/builds/69353#01982dca-4b67-445a-a4fa-121fcfc24aaa | FLAKY         | 20/21    | upstream reliability is '98.4478935698448'. current run reliability is '95.23809523809523'. drift is 3.2098 and the allowed drift is set to 50. The test should PASS    |
| ControllerLogLimitMirrorMakerTests | test_mirror_maker_with_limits |                                                                                                                                        | ducktape    | https://buildkite.com/redpanda/redpanda/builds/69353#01982dc5-77f5-4a50-8a56-ffbe371b2db3 | FLAKY         | 20/21    | upstream reliability is '98.77551020408163'. current run reliability is '95.23809523809523'. drift is 3.53741 and the allowed drift is set to 50. The test should PASS  |
| RandomNodeOperationsTest           | test_node_operations          | {"cloud_storage_type": 2, "compaction_mode": "adjacent_merge", "enable_failures": true, "mixed_versions": true, "with_iceberg": false} | ducktape    | https://buildkite.com/redpanda/redpanda/builds/69353#01982dc5-77f6-42f3-b33a-3596d91318de | FLAKY         | 15/21    | upstream reliability is '99.73614775725594'. current run reliability is '71.42857142857143'. drift is 28.30758 and the allowed drift is set to 50. The test should PASS |
</details>
<!--/body-->

Lazin committed:
ts: Use scheduling group to do disk I/O

Currently, the ts_read_sg consumer group is only used for segment
hydration. This allows the reader that consumes from the cloud storage
cache to saturate the disk t-put. This commit fixes that by shifting
I/O operations to also run under the scheduling group.

Signed-off-by: Evgeny Lazin <4lazin@gmail.com>

Lazin commented:
> I think this is because here we are just opening the streams in the `ts_read` group but really what we need to do is do the dma_reads in the `ts_read` group? I don't think the creation matters really.

looks like this is the case, I moved the I/O to the SG

StephanDollberg reviewed (approved):
Yep this is looking better now.

vbotbuildovich commented:
/backport v25.2.x
